src/httpfetch.h

```diff
@@ -23,10 +23,17 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "util/string.h"
 #include "config.h"
 
-// Can be used in place of "caller" in asynchronous transfers to discard result
-// (used as default value of "caller")
+// These can be used in place of "caller" in to specify special handling.
+// Discard result (used as default value of "caller").
 #define HTTPFETCH_DISCARD 0
+// Indicates that the result should not be discarded when performing a
+// synchronous request (since a real caller ID is not needed for synchronous
+// requests because the result does not have to be retrieved later).
 #define HTTPFETCH_SYNC 1
+// Print response body to console if the server returns an error code.
+#define HTTPFETCH_PRINT_ERR 2
+// Start of regular allocated caller IDs.
+#define HTTPFETCH_CID_START 3
```

Comment:
Is there a particular reason why it's not an `enum`? A new type would help to tell the variables apart (u64 vs enum)

Reply:
These can be used interchangeably with the `u64` so making them an enum would require changing the other types too, wouldn't it?

Reply:
Ah, and indexing arrays with an enum tends to cause problems on MSVC. Fair enough.